string device
float32 value
